pkg/middlewares/aws_defaults.go

```diff
@@ -0,0 +1,128 @@
+	resourcesToIgnore := make([]resource.Resource, 0)
+
+	for _, remoteResource := range remoteResources {
+		// Ignore all resources other than iam role
+		if remoteResource.TerraformType() != aws.AwsIamRoleResourceType {
+			continue
+		}
+
+		if match := strings.HasPrefix(*remoteResource.(*aws.AwsIamRole).Path, defaultIamRolePathPrefix); match {
+			resourcesToIgnore = append(resourcesToIgnore, remoteResource)
+		}
+	}
+
+	return resourcesToIgnore
+}
+
+func (m AwsDefaults) awsIamPolicyAttachmentDefaults(remoteResources []resource.Resource) []resource.Resource {
+	resourcesToIgnore := make([]resource.Resource, 0)
+
+	for _, remoteResource := range remoteResources {
+		// Ignore all resources other than iam policy attachment
+		if remoteResource.TerraformType() != aws.AwsIamPolicyAttachmentResourceType {
+			continue
+		}
+
+		defaultRolesCount := 0
+		for _, roleId := range *remoteResource.(*aws.AwsIamPolicyAttachment).Roles {
+			var role *aws.AwsIamRole
+			for _, res := range remoteResources {
+				if res.TerraformType() == aws.AwsIamRoleResourceType && res.TerraformId() == roleId {
+					role = res.(*aws.AwsIamRole)
+					break
+				}
+			}
+
+			if match := strings.HasPrefix(*role.Path, defaultIamRolePathPrefix); match {
+				defaultRolesCount++
+			}
+		}
+
+		// Check if all of the policy's roles are default AWS roles
+		if defaultRolesCount == len(*remoteResource.(*aws.AwsIamPolicyAttachment).Roles) {
+			resourcesToIgnore = append(resourcesToIgnore, remoteResource)
+		}
+	}
+
+	return resourcesToIgnore
+}
+
+func (m AwsDefaults) awsIamRolePolicyDefaults(remoteResources []resource.Resource) []resource.Resource {
+	resourcesToIgnore := make([]resource.Resource, 0)
+
+	for _, remoteResource := range remoteResources {
+		// Ignore all resources other than role policy
+		if remoteResource.TerraformType() != aws.AwsIamRolePolicyResourceType {
+			continue
+		}
+
+		var role *aws.AwsIamRole
+		for _, res := range remoteResources {
+			if res.TerraformType() == aws.AwsIamRoleResourceType && res.TerraformId() == *remoteResource.(*aws.AwsIamRolePolicy).Role {
+				role = res.(*aws.AwsIamRole)
+				break
+			}
+		}
+
+		if match := strings.HasPrefix(*role.Path, defaultIamRolePathPrefix); match {
+			resourcesToIgnore = append(resourcesToIgnore, remoteResource)
+		}
+	}
+
+	return resourcesToIgnore
+}
+
+func (m AwsDefaults) Execute(remoteResources, resourcesFromState *[]resource.Resource) error {
+	newRemoteResources := append([]resource.Resource{}, *remoteResources...)
+	newResourcesFromState := append([]resource.Resource{}, *resourcesFromState...)
+	resourcesToIgnore := make([]resource.Resource, 0)
+
+	resourcesToIgnore = append(resourcesToIgnore, m.awsIamRoleDefaults(*remoteResources)...)
+	resourcesToIgnore = append(resourcesToIgnore, m.awsIamPolicyAttachmentDefaults(*remoteResources)...)
+	resourcesToIgnore = append(resourcesToIgnore, m.awsIamRolePolicyDefaults(*remoteResources)...)
+
+	for _, resourceToIgnore := range resourcesToIgnore {
```

Comment:
Why not using the same logic than the others middlewares ? Rather than removing element from the slice, append them if they are not ignored to a new slice and replace the original one. WDYT @wbeuil @moadibfr 

If we choose to remove items why mutate the original slice directly ?

Reply:
Note that we cannot perform indexing on a slice pointer, that's why a copied the original slice

Reply:
The problem here is that we perform a loop for each resource to ignore. That means the loop can pass multiple times on the same element. If we used the same logic than the others middlewares, it would append resources more than once and break the scan result. But may be you had something better in mind?